Code/Engine/RendererCore/Font/TextData.h

```diff
@@ -0,0 +1,360 @@
+
+  /**	Returns the height of a line in pixels. */
+  EZ_RENDERERCORE_DLL ezUInt32 GetLineHeight() const;
+
+  /**	Gets information describing a single line at the specified index. */
+  EZ_RENDERERCORE_DLL const ezTextLine& GetLine(ezUInt32 idx) const { return m_Lines[idx]; }
+
+  /**	Returns font texture for the provided page index.  */
+  EZ_RENDERERCORE_DLL const ezTexture2DResourceHandle& GetTextureForPage(ezUInt32 page) const;
+
+  /**	Returns the number of quads used by all the characters in the provided page. */
+  EZ_RENDERERCORE_DLL ezUInt32 GetNumQuadsForPage(ezUInt32 page) const { return m_PageInfos[page].m_NumQuads; }
+
+  /**	Returns the width of the actual text in pixels. */
+  EZ_RENDERERCORE_DLL ezUInt32 GetWidth() const;
+
+  /**	Returns the height of the actual text in pixels. */
+  EZ_RENDERERCORE_DLL ezUInt32 GetHeight() const;
+
+private:
+  /**
+		 * Copies internally stored data in temporary buffers to a persistent buffer.
+		 *
+		 * @param[in]	text			Text originally used for creating the internal temporary buffer data.
+		 * @param[in]	buffer			Memory location to copy the data to. If null then no data will be copied and the
+		 *								parameter @p size will contain the required buffer size.
+		 * @param[in]	size			Size of the provided memory buffer, or if the buffer is null, this will contain the
+		 *								required buffer size after method exists.
+		 * @param[in]	freeTemporary	If true the internal temporary data will be freed after copying.
+		 *
+		 * @note	Must be called after text data has been constructed and is in the temporary buffers.
+		 */
+  EZ_RENDERERCORE_DLL void GeneratePersistentData(const ezString& text, bool freeTemporary = true);
+
+private:
+  friend class ezTextLine;
+
+  /**	Returns Y offset that determines the line on which the characters are placed. In pixels. */
+  ezInt32 GetBaselineOffset() const;
+
+  /**	Returns the width of a single space in pixels. */
+  ezUInt32 GetSpaceWidth() const;
+
+  /** Gets a description of a single character referenced by its sequential index based on the original string. */
+  const ezFontGlyph& GetCharacter(ezUInt32 idx) const { return *m_Chars[idx]; }
+
+  /** Gets a description of a single word referenced by its sequential index based on the original string. */
+  const ezTextWord& GetWord(ezUInt32 idx) const { return m_Words[idx]; }
+
+private:
+  ezDynamicArray<const ezFontGlyph*> m_Chars;
+  ezUInt32 m_NumChars;
+
+  ezDynamicArray<ezTextWord> m_Words;
+  ezUInt32 m_NumWords;
+
+  ezDynamicArray<ezTextLine> m_Lines;
+  ezUInt32 m_NumLines;
+
+  ezDynamicArray<ezPageInfo> m_PageInfos;
+  ezUInt32 m_NumPageInfos;
+
+  ezFontResourceHandle m_Font;
+  const ezFontBitmap* m_FontBitmap;
+
+  // Static buffers used to reduce runtime memory allocation
+protected:
+  /** Stores per-thread memory buffers used to reduce memory allocation. */
+  // Note: I could replace this with the global frame allocator to avoid the extra logic
```

Comment:
That's certainly a good idea. That would get rid of the static thread_local ezBufferData* MemBuffer right ? Because I don't like that variable.